planetaryimage/pds3image.py

```diff
@@ -138,7 +138,7 @@ def _save(self, file_to_write, overwrite):
 
         if self._sample_bytes != self.data.itemsize:
             self.label['IMAGE']['SAMPLE_BITS'] = self.data.itemsize * 8
-            sample_type_to_save = self.DTYPES[self.dtype.byteorder + self.dtype.char]
+            sample_type_to_save = self.DTYPES[self._sample_type[0] + self.dtype.kind]
```

Comment:
1. When we modify numpy array value, it changes its `byte order` to `native (=)` from `>` or `<`. Which in case looses its endianness. By using `self._sample_type` we can preserve the endianness of the input data.

2. Using `dtype.kind` instead of `dtype.char` now.
```
dtype.kind	A character code (one of ‘biufcOSUV’) identifying the general kind of data.
dtype.char	A unique character code for each of the 21 different built-in types.
```